pkg/controller/egress/controller.go

```diff
@@ -426,6 +427,7 @@ func (c *EgressController) updateEgress(old, cur interface{}) {
 		groupSelector := antreatypes.NewGroupSelector("", curEgress.Spec.AppliedTo.PodSelector, curEgress.Spec.AppliedTo.NamespaceSelector, nil, nil)
 		c.groupingInterface.AddGroup(egressGroupType, curEgress.Name, groupSelector)
 	}
+	c.updateEgressStatusCondition(cur, egressv1alpha2.IPAllocated, c.ipAllocationMap[curEgress.Name] != nil)
```

Comment:
ditto, it shouldn't make external call here.
And accessing `c.ipAllocationMap` is not thread-safe here. It could be written by workers at any time. I think you can know whether an IP is allocated in `syncEgressIP`,